### ISSUE TITLE
Out of memory during 5.0 ng serve --aot

### ISSUE THREAD

**Issue by scein** [angular/angular#20138]
ng serve --aot sometimes cause node 8 to run out of stack space. I've tried with --max-old-space-size=8192 (my mem size) without succeess.

node 8.7
npm 5.5.1
angular 5.0.0

**Comment by Toxicable** [comment]
Thanks for reporting this issue. However, you didn't provide sufficient information for us to understand and reproduce the problem. Please check out [our submission guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-submitting-an-issue) to understand why we can't act on issues that are lacking important information.

If the problem still persists, please file a new issue and ensure you provide all of the required information when filling out the issue template.

**Comment by scein** [comment]
You're welcome. Hopefully this gets fixed soon.

**Comment by scein** [comment]
What more information is need? I run ng server --aot on a moderate project, and it runs out of memory before completing...

**Comment by mlc-mlapis** [comment]
@scein ... be serious ... `a moderate project` ... what is it? It could be anything and we already saw very strange brain waves ...

**Comment by scein** [comment]
@mlc-mlapis about 40 components over three modules. It uses bootstrap4.

**Comment by scein** [comment]
Why be so condescending to people reporting bugs. I'm new to all this.

**Comment by icepeng** [comment]
https://github.com/angular/angular-cli/issues/5618

**Comment by scein** [comment]
@icepeng thanks. but i'm not building, i'm doing serve --aot. Basically the same thing?

**Comment by mlc-mlapis** [comment]
@scein ... but it could be just one small and atypical thing ... maybe even not related to Angular at all ... so just do some basic eliminations to get what is the problematic part ... then we will see ...

**Comment by scein** [comment]
@mlc-mlapis what do you need to know?

**Comment by mlc-mlapis** [comment]
@scein ... I? Nothing ... just invest some time to be more exact in a moment when you express some judges publicly. It is very easy to say only ... `I have a problem! Help me!` .... we can see it in many issues here each day.

**Comment by icepeng** [comment]
If you want to get a good answer, follow the guidelines.

About your issue, build and serve is basically same. However, the memory issue is more related to angular-cli. Read the link that I posted and If there is no way to solve it there, then post a comment under that thread.

**Comment by scein** [comment]
Fine, I used the template.

**Comment by rdzidziguri** [comment]
@scein Sorry but 40 components in 3 modules this can not even be considered as medium size app. I have seen that behavior only once or twice and there was like 25 modules and over 350 components. I agree with others u should investigate the root of the problem, but just in case there is something u justr want to have fixed try out this package.

https://www.npmjs.com/package/increase-memory-limit

**Comment by ronbegley** [comment]
For what it's worth, upgrading Nodejs from 6.9.x  (where I got the exact same error you described) to 8.9.0 LTS solved the issue for me
